python/llm/src/bigdl/llm/transformers/models/llama.py

```diff
@@ -106,6 +106,16 @@ def llama_attention_forward_4_31(
 ) -> Tuple[torch.Tensor, Optional[torch.Tensor], Optional[Tuple[torch.Tensor]]]:
     bsz, q_len, _ = hidden_states.size()
     device = hidden_states.device
+    # for flash attention
+    original_dtype = hidden_states.dtype
+    if not self.training and not hidden_states.requires_grad:
+        fsdp_flag = check_flash_attention_available(hidden_states)
```

Comment:
shall we make `fsdp_flag` a global variable?

Reply:
As `fsdp_flag` depends on input device, and it may vary during runtime, global variable seems a little dangerous?

Reply:
I have verifed locally, if we make `fsdp_flag` global, for below test case, it will raise error of `F.scaled_dot_product_attention` at the second generate:
```python
model = model.to('xpu')
model.generate(xxx)

model = model.to('cpu')
model.generate(xxx)
```
And global variable seems not bring obvious performance change.
If we don't make totally isolated env for cpu and gpu, I prefer to keep current usage.
How do you think of it?

Reply:
ok